tutorials/scripting/gdextension/gdextension_cpp_example.rst

```diff
@@ -0,0 +1,652 @@
+would need to showcase a far more complete example.
+
+This is the required syntax:
+
+.. code-block:: C++
+
+    some_other_node->connect("the_signal", this, "my_method");
+
+Note that you can only call ``my_method`` if you've previously registered it in
+your ``_bind_methods`` method.
+
+Having your object sending out signals is more common. For our wobbling
+Godot icon, we'll do something silly just to show how it works. We're going to
+emit a signal every time a second has passed and pass the new location along.
+
+In our ``gdexample.h`` header file, we need to define a new member ``time_emit``:
+
+.. code-block:: C++
+
+    ...
+        float time_passed;
+        float time_emit;
+        float amplitude;
+    ...
+
+This time, the changes in ``gdexample.cpp`` are more elaborate. First,
+you'll need to set ``time_emit = 0.0;`` in either our ``_init`` method or in our
+constructor. We'll look at the other 2 needed changes one by one.
+
+In our ``_bind_methods`` method, we need to declare our signal. This is done
+as follows:
+
+.. code-block:: C++
+
+    void GDExample::_bind_methods() {
+        ClassDB::add_property("GDExample", PropertyInfo(Variant::FLOAT, "amplitude"), "set_amplitude", "get_amplitude");
+        ClassDB::add_property("GDExample", PropertyInfo(Variant::FLOAT, "speed"), "set_speed", "get_speed");
+
+
+        ADD_SIGNAL(MethodInfo("position_changed", PropertyInfo(Variant::OBJECT, "node"), PropertyInfo(Variant::VECTOR2, "new_pos")));
+    }
+
+Here, our ``ADD_SIGNAL`` method can be a single call first taking the
+signals name, then having pairs of the type specifying the parameter name and
+the value of each parameter we'll send along with this signal.
+
+Next, we'll need to change our ``_process`` method:
+
+.. code-block:: C++
+
+    void GDExample::_process(float delta) {
+        time_passed += speed * delta;
+
+        Vector2 new_position = Vector2(
+            amplitude + (amplitude * sin(time_passed * 2.0)),
+            amplitude + (amplitude * cos(time_passed * 1.5))
+        );
+
+        set_position(new_position);
+
+        time_emit += delta;
+        if (time_emit > 1.0) {
+            emit_signal("position_changed", this, new_position);
+
+            time_emit = 0.0;
+        }
+    }
+
+After a second has passed, we emit our signal and reset our counter. We can add
+our parameter values directly to ``emit_signal``.
+
+Once the GDExtension library is compiled, we can go into Godot and select our sprite
+node. In the **Node** dock, we can find our new signal and link it up by pressing
+the **Connect** button or double-clicking the signal. We've added a script on
+our main node and implemented our signal like this:
+
+.. code-block:: GDScript
+
+    extends Node
+
+    func _on_Sprite2D_position_changed(node, new_pos):
+        print("The position of " + node.get_class() + " is now " + str(new_pos))
+
+Every second, we output our position to the console.
+
```

Comment:
Similar to what was discussed on properties, we really need a follow up page that goes more in depth on signals and specifically how to connect to signals from GDExtensions as the callables syntax is not something one would easily figure out on their own. But that should be a follow up PR